purchases/src/main/kotlin/com/revenuecat/purchases/BillingWrapper.kt

```diff
@@ -234,21 +257,37 @@ internal class BillingWrapper internal constructor(
         )
     }
 
-    fun consumePurchase(token: String) {
+    fun consumePurchase(
+        token: String,
+        onConsumed: (responseCode: Int, purchaseToken: String) -> Unit
+    ) {
         debugLog("Consuming purchase with token $token")
         executeRequestOnUIThread { connectionError ->
             if (connectionError == null) {
-                billingClient?.consumeAsync(token) { _, _ -> }
+                billingClient?.consumeAsync(token) { responseCode, purchaseToken -> onConsumed(responseCode, purchaseToken) }
             }
         }
     }
 
+    fun queryPurchases(@SkuType skuType: String): Purchase.PurchasesResult? {
+        return billingClient?.let {
+            debugLog("[QueryPurchases] Querying $skuType")
+            it.queryPurchases(skuType)
+        }
+    }
+
     override fun onPurchasesUpdated(@BillingClient.BillingResponse responseCode: Int, purchases: List<Purchase>?) {
         if (responseCode == BillingClient.BillingResponse.OK && purchases != null) {
-            purchases.forEach {
-                debugLog("BillingWrapper purchases updated: ${it.toHumanReadableDescription()}")
+            purchases.map { purchase ->
+                debugLog("BillingWrapper purchases updated: ${purchase.toHumanReadableDescription()}")
+                var type: String?
+                synchronized(this@BillingWrapper) {
+                    type = productTypes[purchase.sku]
```

Comment:
Is it possible that type isn't in this dict?

Reply:
If the purchase has been made through the play store instead of through our makePurchase method